client/web/src/insights/InsightsRouter.tsx

```diff
@@ -40,6 +44,10 @@ export interface InsightsRouterProps
 export const InsightsRouter = withAuthenticatedUser<InsightsRouterProps>(props => {
     const { match, ...outerProps } = props
 
+    const settings = !isErrorLike(outerProps.settingsCascade.final) ? outerProps.settingsCascade.final : {}
+    const experimentalFeatures: SettingsExperimentalFeatures = settings?.experimentalFeatures ?? {}
+    const isDashboardsEnabled = experimentalFeatures.codeInsightsDashboards
```

Comment:
Why do we rename the setting here? 

It's easier to follow the data flow when things are not renamed in the middle. Then if a developer is aware of the `codeInsightsDashboards` experimental setting, he can understand it anywhere in the codebase.

```suggestion
    const codeInsightsDashboards = experimentalFeatures.codeInsightsDashboards
```